File: C/RangeSumQueryImmutable.c
#include <stdlib.h>

typedef struct
{
	int* prefixsum;
} NumArray;

NumArray* numArrayCreate(int* nums, int numsSize)
{
	NumArray* arr = (NumArray*)malloc(sizeof(NumArray));
	arr->prefixsum = (int*)malloc(sizeof(int) * (numsSize + 1));
	for (int i = 0; i < numsSize; ++i)
		arr->prefixsum[i + 1] = arr->prefixsum[i] + nums[i];
	return arr;
}

int numArraySumRange(NumArray* obj, int i, int j)
{
	return obj->prefixsum[j + 1] - obj->prefixsum[i];
}

void numArrayFree(NumArray* obj)
{
	if (obj)
	{
		if (obj->prefixsum)
			free(obj->prefixsum);
		free(obj);
	}
}

/**
 * Your NumArray struct will be instantiated and called as such:
 * NumArray* obj = numArrayCreate(nums, numsSize);
 * int param_1 = numArraySumRange(obj, i, j);
 
 * numArrayFree(obj);
*/

File: python/RangeSumQueryImmutable.py
from typing import List


class NumArray:
    def __init__(self, nums: List[int]):
        self.arr = [0]
        for n in nums:
            self.arr.append(self.arr[-1] + n)

    def sumRange(self, i: int, j: int) -> int:
        return self.arr[j + 1] - self.arr[i]

# Your NumArray object will be instantiated and called as such:
# obj = NumArray(nums)
# param_1 = obj.sumRange(i,j)
